docs/AUTHENTICATION.md

```diff
@@ -0,0 +1,162 @@
+# Authenticating to Google Cloud APIs
+
+`provider-gcp` requires credentials to be provided in order to authenticate to
+the Google Cloud APIs. This can be done in one of the following ways:
+
+- Authenticating using a base-64 encoded service account key in a Kubernetes
+  `Secret`. This is described in detail [here](https://crossplane.io/docs/v1.6/getting-started/install-configure.html#get-gcp-account-keyfile).
+- Authenticating using [Workload Identity](https://cloud.google.com/kubernetes-engine/docs/concepts/workload-identity).
+  This is described in the [section below](#authenticating-with-workload-identity).
+
+## Authenticating with Workload Identity
+
+*Note: This method is supported in `provider-gcp` v0.20.0 and later.*
+
+Using Workload Identity requires some additional setup.
+Many of the steps can also be found in the [documentation](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity).
+
+### Steps
+
+These steps assume you already have a running GKE cluster which has already
+enabled Workload Identity and has a sufficiently large node pool.
+
+Note that you can specify any valid strings to the variables below unless the
+variable is explicitly assigned.
+
+#### 1. Install Crossplane
+
+Install Crossplane from `stable` channel:
+
+```bash
+$ helm repo add crossplane-stable https://charts.crossplane.io/stable
+$ helm install crossplane --create-namespace --namespace crossplane-system crossplane-stable/crossplane
+```
+
+`provider-gcp` can be installed with either the [Crossplane CLI](https://crossplane.io/docs/v1.6/getting-started/install-configure.html#install-crossplane-cli)
+or a `Provider` resource as below:
+
+```console
+$ cat <<EOF | kubectl apply -f -
+apiVersion: pkg.crossplane.io/v1
+kind: Provider
+metadata:
+  name: ${PROVIDER_GCP}
+spec:
+  package: crossplane/provider-gcp:v${VERSION} # v0.20.0 or later
+  # Set a controllerConfigRef if you have a ControllerConfig:
+  # controllerConfigRef:
+  #   name: ${CONTROLLER_CONFIG}
+EOF
+```
+
+#### 2. Configure service accounts to use Workload Identity
+
+Create a GCP service account, which will be used for provisioning actual
+infrastructure in GCP, and grant IAM roles you need for accessing the Google
+Cloud APIs:
+
+```console
+$ gcloud iam service-accounts create ${GCP_SERVICE_ACCOUNT}
+$ gcloud projects add-iam-policy-binding ${PROJECT_ID} \
+    --member "serviceAccount:${GCP_SERVICE_ACCOUNT}@${PROJECT_ID}.iam.gserviceaccount.com" \
+    --role ${ROLE} \
```

Comment:
nit: 
It would be handy to have all environment variables that the user needs to prepare.

0. Prepare your vars
```
PROJECT_ID=<some-project-id>
GCP_SERVICE_ACCOUNT=<some-sa>
ROLE=<some-role>
```

Reply:
I think the caveat above would be enough?

> Note that you can specify any valid strings to the variables below unless the
 variable is explicitly assigned.

Because each variable needs a description for what the variable means and when it is used, and it will be duplications if we add them in step 0.

Reply:
Hmm, maybe we can add right before they were used like [this](https://github.com/crossplane/provider-aws/blame/master/AUTHENTICATION.md#L43).

But as mentioned, this is just nitpicking, and let me know if you don't want to add.

Reply:
OK, that also makes sense to me. Just added it at [`12ebbd5` (#414)](https://github.com/crossplane/provider-gcp/pull/414/commits/12ebbd5a769c5a3fef90919c8971a13119ca6c1d). Feel free to give me any other feedback!

Reply:
Looking great, thanks!